authentication/migrations/0005_authuser_alter_customuser_user_id.py

```diff
@@ -0,0 +1,40 @@
+# Generated by Django 4.1 on 2024-02-21 10:56
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('authentication', '0004_authuser_companies_remove_customuser_id_and_more'),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='AuthUser',
+            fields=[
+                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('last_login', models.DateTimeField(blank=True, null=True, verbose_name='last login')),
+                ('email', models.EmailField(default=None, max_length=100, unique=True)),
+                ('password', models.CharField(default=None, max_length=128)),
+                ('first_name', models.CharField(default=None, max_length=100)),
+                ('surname', models.CharField(default=None, max_length=100)),
+                ('position', models.CharField(default=None, max_length=25)),
+                ('company_id', models.IntegerField(default=None)),
+                ('user_id', models.IntegerField(default=None)),
+                ('is_startup', models.BooleanField(default='undefined')),
+                ('is_verified', models.BooleanField(default='undefined')),
```

Comment:
I checked this, but there were no complications.

Reply:
It works now, but you can't guarantee correct behavior under unpredictable conditions

Reply:
Now you plan Boolean fields. How are you gonna set string value?

Reply:
Here's the example of empty AuthUser returned when no token was provided:

{'_state': <django.db.models.base.ModelState object at 0x00000267229D6190>, 'id': None, 'last_login': None, 'email': None, 'password': None, 'first_name': None, 'surname': None, 'position': None, 'company_id': None, 'user_id': None, 'is_startup': 'undefined', 'is_verified': 'undefined', 'is_superuser': False, 'is_authenticated': False, 'error': <class 'forum.errors.Error.NO_HEADER'>}

Reply:
Even if you do not currently store the data of this field in the database, it may be used in future development. And data processing in it will be calculated as data of the Boolean type, and not as a string. Consider using this model in the future

Reply:
I need to think of that issue then

Reply:
I have changed concept a little. Here's the changing I've made:
1. CustomUser is authentication model now. If company is related to user, it is added to model instance. The corresponding fields 'company' added, but they aren't taking part in interactions with database in case one needed. Methods added to retrieve company type and incapsulate database interactions. 
2. Company and UserAndCompanyRelation models are moved to Authentication app to avoid circular import. All the related imports corrected accordingly.
3. Permissions rewrited and simplified.
4. Authentication class simplified. Errors are handling with Exception raising instead of pass them into request.user instance.
5. Docstrings added